src/utils/hooks/useDataViews.tsx

```diff
@@ -42,35 +44,58 @@ export const useDataViewsList = () => {
   } = useUserState()
 
   useEffect(() => {
-    processGetRequest(getServerUrl('dataViews'), setError, setLoading, setDataViewsList)
+    processRequest({
+      url: getServerUrl('dataViews'),
+      setError,
+      setLoading,
+      setStateMethod: setDataViewsList,
+    })
   }, [templatePermissions])
 
   return { error, loading, dataViewsList }
 }
 
 export const useDataViewsTable = ({ dataViewCode, apiQueries, filter }: DataViewTableProps) => {
+  const { strings } = useLanguageProvider()
   const [error, setError] = useState<ErrorResponse | null>(null)
   const [loading, setLoading] = useState(false)
   const [dataViewTable, setDataViewTable] = useState<DataViewsTableResponse>()
+  const [filterDefinitions, setFilterDefinitions] = useState<any>()
   const {
     userState: { templatePermissions },
   } = useUserState()
 
+  const updateDataViewTable = (newDataViewTable: DataViewsTableResponse) => {
+    // We only want to load filterDefinitions once (per dataView code),
+    // otherwise it gets into an infinite loop. Similarly, we don't want to set
+    // the dataViewTable data until we've loaded filterDefinitions, or the Table
+    // page will display the full list briefly before showing the filtered list.
+
+    // TO-DO: Fetch filter definitions in a separate query *before* requesting
+    // dataViewTable
```

Comment:
The filter definitions are received in the same query/response as the actual table data.

This is not ideal, as when loading a page with filters already applied (in url), we have to fetch the full (unfiltered) data just to get the filter definitions (i.e. how to display the current url filters), and then fetch the filtered data again after that.

I've got this little workaround here to "hide" the re-fetch (i.e. don't render table until filter definitions are loaded), but it'd be better to make a separate endpoint just to query filter definitions *before* querying the table data (with filters applied). 

The only reason I didn't do that already is because it's slightly more efficient in the back-end to build both table columns and filter definitions in one go, since they share a lot of the same database queries and depend on each other to some extent. 

Something to think about for later...